buildSrc/src/main/kotlin/codegen/math/Utils.kt

```diff
@@ -0,0 +1,200 @@
+		.receiver(type)
+		.apply(block)
+		.build()
+		.also { addFunction(it) }
+}
+
+@KotlinPoetDsl
+inline fun FileSpec.Builder.buildClass(
+	name: String,
+	block: TypeSpec.Builder.() -> Unit = {}
+): TypeSpec {
+	return TypeSpec.classBuilder(name)
+		.apply(block)
+		.build()
+		.also { addType(it) }
+}
+
+@KotlinPoetDsl
+inline fun FileSpec.Builder.buildClass(
+	className: ClassName,
+	block: TypeSpec.Builder.() -> Unit = {}
+): TypeSpec {
+	return TypeSpec.classBuilder(className)
+		.apply(block)
+		.build()
+		.also { addType(it) }
+}
+
+@KotlinPoetDsl
+fun TypeSpec.Builder.modifiers(vararg modifiers: KModifier) {
+	addModifiers(*modifiers)
+}
+
+@KotlinPoetDsl
+inline fun TypeSpec.Builder.primaryConstructor(
+	block: FunSpec.Builder.() -> Unit
+): FunSpec {
+	return FunSpec.constructorBuilder()
+		.apply(block)
+		.build()
+		.also { primaryConstructor(it) }
+}
+
+@KotlinPoetDsl
+inline fun TypeSpec.Builder.secondaryConstructor(
+	block: FunSpec.Builder.() -> Unit
+): FunSpec {
+	return FunSpec.constructorBuilder()
+		.apply(block)
+		.build()
+		.also { addFunction(it) }
+}
+
+@KotlinPoetDsl
+inline fun TypeSpec.Builder.property(
+	name: String,
+	type: TypeName,
+	vararg modifiers: KModifier,
+	block: PropertySpec.Builder.() -> Unit = {}
+): PropertySpec {
+	return PropertySpec.builder(name, type, *modifiers)
+		.apply(block)
+		.build()
+		.also { addProperty(it) }
+}
+
+@KotlinPoetDsl
+fun PropertySpec.Builder.modifiers(vararg modifiers: KModifier) {
+	addModifiers(*modifiers)
+}
+
+@KotlinPoetDsl
+inline fun PropertySpec.Builder.getter(
+	block: FunSpec.Builder.() -> Unit
+): FunSpec {
+	return FunSpec.getterBuilder()
+		.apply(block)
+		.build()
+		.also { getter(it) }
+}
+
+@KotlinPoetDsl
+inline fun TypeSpec.Builder.mutableProperty(
+	name: String,
+	type: TypeName,
+	vararg modifiers: KModifier,
+	block: PropertySpec.Builder.() -> Unit = {}
+): PropertySpec {
+	return PropertySpec.builder(name, type, *modifiers)
+		.mutable(true)
+		.apply(block)
+		.build()
+		.also { addProperty(it) }
+}
+
+@KotlinPoetDsl
+inline fun PropertySpec.Builder.setter(
+	block: FunSpec.Builder.() -> Unit
+): FunSpec {
+	return FunSpec.setterBuilder()
+		.apply(block)
+		.build()
+		.also { setter(it) }
+}
+
+@KotlinPoetDsl
+inline fun TypeSpec.Builder.function(name: String, block: FunSpec.Builder.() -> Unit): FunSpec {
+	return FunSpec.builder(name)
+		.apply(block)
+		.build()
+		.also { addFunction(it) }
+}
+
+@KotlinPoetDsl
+fun FunSpec.Builder.modifiers(vararg modifiers: KModifier) {
+	addModifiers(*modifiers)
+}
+
+@KotlinPoetDsl
+inline fun FunSpec.Builder.parameter(
+	name: String,
+	type: TypeName,
+	vararg modifiers: KModifier,
+	block: ParameterSpec.Builder.() -> Unit = {}
+): ParameterSpec {
+	return ParameterSpec.builder(name, type, *modifiers)
+		.apply(block)
+		.build()
+		.also { addParameter(it) }
+}
+
+@KotlinPoetDsl
+inline fun FunSpec.Builder.controlFlow(controlFlow: String, block: FunSpec.Builder.() -> Unit): FunSpec {
```

Comment:
I'm a bit sceptical about this helper function. It builds the `FunSpec`, instead of just providing control flow as the name suggests.

Reply:
I'm not sure what you mean. The DSL builder function is just a wrapper around the already existing controlFlow mechanism, which builds a FunSpec. Kotlin Poet does not provide a ControlFlowSpec.

Reply:
I'm saying the function should return `Unit` and shouldn't build the `FunSpec`.

Reply:
Ooooh. I didn't realize I was building the outer FunSpec with the build call. I was under the impression that `beginControlFlow` returned a control flow as a FunSpec. Will be fixed in next update 👌

Reply:
https://github.com/Dominaezzz/kgl/pull/19/commits/a3db960c6005dde1b2ae4c3b557435bccad82cff